### PR TITLE
Use GitHub as a source of the plugin's documentation on plugins.jenkins.io

### PR DESCRIPTION
**Pull request by zhaoying818**
subj.
https://issues.jenkins-ci.org/browse/JENKINS-59172

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <packaging>hpi</packaging>
     <version>${revision}${changelist}</version>
     <name>Text Finder</name>
-    <url>https://wiki.jenkins.io/display/JENKINS/Text-finder+Plugin</url>
+    <url>https://github.com/jenkinsci/text-finder-plugin</url>
```
Comment (basil) [inline pom.xml]: Can we avoid hard-coding the name of the plugin by relying on `artifactId` instead? That is how it is done in the archetype as well:

https://github.com/jenkinsci/archetypes/blob/8a49c0c784267d8baf3841199fa4292aad18225a/empty-plugin/src/main/resources/archetype-resources/pom.xml#L26

```suggestion
    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```

File: README.md [hunk 2]
```diff
@@ -5,21 +5,90 @@
...
+a non-zero exit status, but the reclassification only applies to builds
+which returned an overall exit status of zero.
 
-See [the Text Finder Plugin page](https://wiki.jenkins-ci.org/display/JENKINS/Text-finder+Plugin)
-on the Jenkins wiki for more information.
+### Usage
+
+The basic pipeline syntax is as follows:
+
+        findText regexp: '<regular expression>', fileSet: '<file set>'
```
Comment (basil) [inline README.md]: Here and elsewhere, should we surround quoted code with [triple-backticks](https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax#quoting-code)?

File: README.md [hunk 3]
```diff
@@ -5,21 +5,90 @@
...
+
+        findText regexp: '<regular expression>', alsoCheckConsoleOutput: true
+
+To customize the build result, you can use any combination of
+`succeedIfFound`, `unstableIfFound`, or `notBuiltIfFound`. For example,
+to mark the build as unstable if the expression is found:
+
+        findText regexp: '<regular expression>, alsoCheckConsoleOutput: true, unstableIfFound: true
+
+### Changelog
```
Comment (basil) [inline README.md]: I think I'd prefer to keep the changelog separate from the main `README`. Could you please split this out into a separate `CHANGELOG.md` following the example in e.g jenkinsci/workflow-job-plugin#144?

Comment (zhaoying818) [reply README.md]: this suggestion is good!

File: README.md (hunk 2 above)
Comment (zhaoying818) [reply README.md]: I think we should.

**Pull request merged**